pkg/trace/agent/agent.go

```diff
@@ -150,15 +147,18 @@ func (a *Agent) loop() {
 
 // Process is the default work unit that receives a trace, transforms it and
 // passes it downstream.
-func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalculator) {
+func (a *Agent) Process(p *api.Payload) {
 	if len(p.Traces) == 0 {
 		log.Debugf("Skipping received empty payload")
 		return
 	}
 	defer timing.Since("datadog.trace_agent.internal.process_payload_ms", time.Now())
 	ts := p.Source
 	ss := new(writer.SampledSpans)
-	sinputs := make([]stats.Input, 0, len(p.Traces))
+	var sinputs []stats.Input
+	if !p.ClientComputedStats {
+		sinputs = make([]stats.Input, 0, len(p.Traces))
+	}
```

Comment:
I think you can completely drop this. It is fine to append to a slice which hasn't been "made"

Reply:
I think it shouldn't be dropped: it will generate a lot of reallocations: trace payloads can have many traces and slices are reallocated by x2 (2,4,8,16 ... 
The cost of this allocating this array at the right size is already visible in some of my profiles

Reply:
That's true, good point. In that case maybe we can move it into the other `if !p.ClientComputedStats {` clause. No need to have them so far away from each other. It's inviting bugs :)